radiography/src/main/java/radiography/ScanExecutors.kt

```diff
@@ -0,0 +1,82 @@
+package radiography
+
+import android.os.Handler
+import android.os.Looper
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+public object ScanExecutors {
+
+  internal val mainHandler by lazy {
+    Handler(Looper.getMainLooper())
+  }
+
+  /**
+   * Runs the scanning tasks synchronously without any thread check, posting or exception catching.
+   */
+  @JvmField
+  public val PassthroughExecutor: ScanExecutor = ScanExecutor { callable ->
+    callable.call()
+  }
+
+  /**
+   * Executes the scanning callbacks on the provided [delegate] ScanExecutor but catches any
+   * exception and returns an error string instead.
+   */
+  @JvmStatic
+  public fun NeverThrowingExecutor(delegate: ScanExecutor): ScanExecutor = ScanExecutor { callable ->
+    try {
+      delegate.execute(callable)
+    } catch (throwable: Throwable) {
+      "Exception when scanning: ${throwable.message}"
+    }
+  }
+
+  /**
+   * Runs the scanning tasks synchronously. If the current thread is the same as the
```

Comment:
If posted to another handler these aren't run synchronously? Is this comment out of date?